rmw_fastrtps_cpp/src/publisher.cpp

```diff
@@ -106,6 +106,13 @@ rmw_fastrtps_cpp::create_publisher(
   }
   auto cleanup_info = rcpputils::make_scope_exit(
     [info, participant]() {
+      if (info->publisher_) {
+        if (!Domain::removePublisher(info->publisher_)) {
```

Comment:
it would be better to add another `scope_exit` call after:
https://github.com/ros2/rmw_fastrtps/blob/7944a2cfeddbfc197e1d44e55089d38cb4594d56/rmw_fastrtps_cpp/src/publisher.cpp#L164-L171
instead of adding the cleanup here.